main.cpp

```diff
@@ -19,13 +19,32 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& _rewriter) : rewriter_(_rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
-    }
+	void run(const MatchFinder::MatchResult& Result) override {
+		if (auto* cast = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+			auto type = Lexer::getSourceText(
+				CharSourceRange::getTokenRange(
+					cast->getLParenLoc().getLocWithOffset(1),
+					cast->getRParenLoc().getLocWithOffset(-1)),
+				*Result.SourceManager, Result.Context->getLangOpts());
+
+			rewriter_.ReplaceText(
+				CharSourceRange::getCharRange(
+					cast->getLParenLoc(),
+					cast->getSubExprAsWritten()->getBeginLoc()),
+				("static_cast<" + type + ">(").str());
+
+			rewriter_.InsertText(
+				Lexer::getLocForEndOfToken(

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
getLocForEndOfToken() computes the source location **just past the end** of the token at this source location.
If we try to use `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` and the variable we declare in main has name longer than one symbol like `foo` we will get `f)oo`.
If we try to use `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` we will get the same output but function won't get language options `getLangOpts()` from the context as it seems to me.

Reply:
That is correct!
Both options `Lexer::getLocForEndOfToken(styleCastExpr->getEndLoc(),...)` and `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(),...)` can be applied here, they give the same expected result.